modules/blocks/README.md

```diff
@@ -0,0 +1,162 @@
+
+- `name: string` - A human-readable [localized](https://codex.wordpress.org/I18n_for_WordPress_Developers#Handling_JavaScript_files) label for the block. Shown in the block picker.
+- `edit( { attributes: Object, setAttributes: Function } ): WPElement` - Returns an element describing the markup of a block to be shown in the editor. A block can update its own state in response to events using the `setAttributes` function, passing an object of properties to be applied as a partial update.
+- `save( { attributes: Object } ): WPElement` - Returns an element describing the markup of a block to be saved in the published content. This function is called before save and when switching to an editor's HTML view.
+- `tagName: string` - An alternative to defining `edit` and `save` behaviors, if passed a [tag name](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement#Parameters), delegates default behavior of an editable field on that node type. Typically used for simple, text-heavy elements (paragraphs, headings) to avoid excess meta storage of text.
+- `controls: string[]` - Slugs for controls to be made available to block. See also: [`wp.blocks.registerControl`](#wpblocksregistercontrol-slug-string-settings-object-)
+
+### `wp.blocks.registerControl( slug: string, settings: Object )`
+
+Registers a new block-level control. Controls appear in a block's toolbar when it receives focus if it is included in the block's `controls` option.
+
+- `icon: string | WPElement` - Slug of the [Dashicon](https://developer.wordpress.org/resource/dashicons/#awards) to be shown in the control's button, or an element if you choose to render your own SVG.
+- `onClick( { attributes: Object, setAttributes: Function } )` - Click behavior for control. Use this to change or toggle an attribute of the block.
+- `isVisible( { attributes: Object } ): boolean` - Called when a block receives focus or changes. Return `false` to prevent the control's button from being shown. If this option is not defined for a control, the button will always be shown.
+- `isActive( { attributes: Object } ): boolean` - Called when a block receives focus or changes. Return `true` to apply an active effect to the control's button, in the case that the control's behavior is a toggle. 
+
+Inline controls for [`Editable`](#editable) elements are identical for every block and cannot be modified.
+
+### `wp.blocks.getBlockSettings( slug: string )`
+
+Returns settings associated with a registered block.
+
+### `wp.blocks.getControlSettings( slug: string )`
+
+Returns settings associated with a registered control.
+
+## Components
+
+Because many blocks share the same complex behaviors, the following components are made available to simplify implementations of your block's `edit` function.
+
+### `Editable`
```

Comment:
- If I understand correctly, if you use the `Editable` component you get the format toolbar for free. This is handy and I'm all for it but, we'll have some issues here: Positioning the toolbar won't be easy to do, also imagine a component uses more than on Editable. From the mockups sometimes the toolbar is shown just above the input, and sometimes it's next to the block controls.